src/tribler/core/upgrade/upgrade.py

```diff
@@ -105,6 +124,9 @@ def run(self):
         self.remove_old_logs()
         self.upgrade_pony_db_14to15()
 
+        if self._db_is_corrupted_exception:
+            raise self._db_is_corrupted_exception  # pylint: disable=raising-bad-type
```

Comment:
Could you please clarify that this line is roughly equivalent to the `system.exit` (if it is correct).